sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/DLPReidentifyText.java

```diff
@@ -0,0 +1,281 @@
+        .apply(
+            "DLPReidentify",
+            ParDo.of(
+                new ReidentifyText(
+                    getProjectId(),
+                    getInspectTemplateName(),
+                    getReidentifyTemplateName(),
+                    getInspectConfig(),
+                    getReidentifyConfig(),
+                    getHeaderColumns())));
+  }
+
+  /** Performs the calls to Cloud DLP service on GCP. */
+  static class ReidentifyText
+      extends DoFn<KV<String, Iterable<Table.Row>>, KV<String, ReidentifyContentResponse>> {
+    private final String projectId;
+    private final String inspectTemplateName;
+    private final String reidentifyTemplateName;
+    private final InspectConfig inspectConfig;
+    private final DeidentifyConfig reidentifyConfig;
+    private transient ReidentifyContentRequest.Builder requestBuilder;
+    private final PCollectionView<List<String>> headerColumns;
+
+    @Setup
+    public void setup() throws IOException {
+      requestBuilder =
+          ReidentifyContentRequest.newBuilder().setParent(ProjectName.of(projectId).toString());
+      if (inspectTemplateName != null) {
+        requestBuilder.setInspectTemplateName(inspectTemplateName);
+      }
+      if (inspectConfig != null) {
+        requestBuilder.setInspectConfig(inspectConfig);
+      }
+      if (reidentifyConfig != null) {
+        requestBuilder.setReidentifyConfig(reidentifyConfig);
+      }
+      if (reidentifyTemplateName != null) {
+        requestBuilder.setReidentifyTemplateName(reidentifyTemplateName);
+      }
+    }
+
+    /**
+     * @param projectId ID of GCP project that should be used for deidentification.
+     * @param inspectTemplateName Template name for inspection. Optional.
+     * @param reidentifyTemplateName Template name for reidentification. Either this or
+     *     reidentifyConfig is required.
+     * @param inspectConfig Configuration object for inspection. Optional.
+     * @param reidentifyConfig Reidentification config containing data transformations. Either this
+     *     or reidentifyTemplateName is required.
+     * @param headerColumns Header row of the table if applicable.
+     */
+    public ReidentifyText(
+        String projectId,
+        String inspectTemplateName,
+        String reidentifyTemplateName,
+        InspectConfig inspectConfig,
+        DeidentifyConfig reidentifyConfig,
+        PCollectionView<List<String>> headerColumns) {
+      this.projectId = projectId;
+      this.inspectTemplateName = inspectTemplateName;
+      this.reidentifyTemplateName = reidentifyTemplateName;
+      this.inspectConfig = inspectConfig;
+      this.reidentifyConfig = reidentifyConfig;
+      this.headerColumns = headerColumns;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext context) throws IOException {
+      try (DlpServiceClient dlpServiceClient = DlpServiceClient.create()) {
```

Comment:
Does it make sense to move the creation of client in startBundle? Just commenting for overall all three use cases.

Reply:
I think we can move it there, it'll always save some memory and cpu cycles.

Reply:
We can move it to startBundle or to setup and close the client in teardown.